### PR TITLE
Fix Jenkins 66025

### PR DESCRIPTION
**Pull request by realitix**
This pull request switch Netty implementation by HTTP implementation.

CI tests should be passed on your CI with your configuration to validate the good behavior.

https://issues.jenkins.io/browse/JENKINS-66025

Thanks

### CODE REVIEW

**Comment by realitix** [comment]
Hello, could you look at this PR ?

Thanks

**Comment by pjdarton** [comment]
It has merge issues that need fixing.

File: src/main/java/io/jenkins/docker/client/DockerAPI.java [hunk 1]
```diff
@@ -236,34 +240,25 @@ public void reallyClose() throws IOException {
...
-                    .withConnectTimeout(connectTimeoutInMillisecondsOrNull);
-            final DefaultDockerClientConfig.Builder configBuilder = new DefaultDockerClientConfig.Builder()
-                    .withDockerHost(dockerUri)
-                    .withCustomSslConfig(toSSlConfig(credentialsId));
-            actualClient = DockerClientBuilder.getInstance(configBuilder)
-                .withDockerCmdExecFactory(cmdExecFactory)
-                .build();
+            DockerHttpClient httpClient = new ApacheDockerHttpClient.Builder().dockerHost(URI.create(dockerUri))
+                    .sslConfig(toSSlConfig(credentialsId)).build();
+            actualClient = DockerClientBuilder.getInstance().withDockerHttpClient(httpClient).build();
```
Comment (pjdarton) [inline src/main/java/io/jenkins/docker/client/DockerAPI.java:253]: Please retail the same "code style" as the previous code.

FYI the reason why it was putting each bit on its own line is so that, in future, if we need to add/remove/change any calls to the builder, all we're doing is adding/removing/changing one line - all the other lines will remain unchanged, and this makes viewing the history in git (e.g. using "git blame") _so_ much easier as you can see where each builder parameter came from.
If we allow multiple calls per line, it means that to add an extra "with..." the code change will show that we've rewritten the entire builder clause rather than just added one extra thing.

TL;DR: Multi-line builder code makes for more maintainable code.

**Review by pjdarton — CHANGES_REQUESTED** [review]
The main codebase has moved on since this PR was first done, and it needs to be rebased on the current latest code before it'll make sense again.

Also, there's a _lot_ of unnecessary changes to files in here.
This PR is a change to the plugin's functionality, so we want cosmetic/layout changes kept to an absolute minimum so that this PR's changes consist of _only_ the functionality changes ... and does so in a manner that "fits in" with the existing code style (however much you might dislike it 😉).
If you find the code intolerably ugly and need to make cosmetic changes to the code's layout etc to make it "prettier", do that in a separate PR that _only_ makes cosmetic code changes.

**Comment by pjdarton** [comment]
@mat1e Has this PR also been superseded by #899 ?

**Comment by mat1e** [comment]
> been

Yes.

**Comment by mat1e** [comment]
> @mat1e Has this PR also been superseded by #899 ?

Yes.
